packages/workflow/src/conversion/image.ts

```diff
@@ -1,22 +1,83 @@
-
-import sharp from "sharp";
-
+import { log } from "@temporalio/activity";
+import { exec as execCallback } from "child_process";
+import fs from "fs";
+import { file } from "tmp-promise";
+import { promisify } from "util";
+const exec = promisify(execCallback);
 
 /**
- * Resizes an image to a maximum height or width
- * @param max_hw 
- * @param format 
- * @returns 
+ * Resizes an image to a maximum height or width using ImageMagick
+ * with progressive loading when supported
+ * @param inputPath Input file path
+ * @param max_hw Maximum height or width
+ * @param format Output format
+ * @param progressive Enable progressive loading for supported formats (defaults to true)
+ * @returns Path to the resized image
  */
-export function imageResizer(max_hw: number, format: keyof sharp.FormatEnum) {
+export async function imageResizer(
+    inputPath: string,
```

Comment:
Do we have a validation somewhere about the input path, for example it must only contain certain characters or within a specific directory? Because it may bring some security concerns, such as https://owasp.org/www-community/attacks/Path_Traversal